app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -18,6 +18,56 @@ $(function () {
     $('#ldap_conn_message').html(msg)
   }
 
+  function ldapConnectionTestPart1 () {
+    var file = $('#dex_connector_ldap_certificate').val()
+    if(!file) {
+      ldapConnectionTestPart2( $('#dex_connector_ldap_current_cert').val() )
+      return;
+    }
+    var fileInput = $('dex_connector_ldap_certificate')[0];
+    var reader = new FileReader();
+
+    reader.onload = function(e) {
+      ldapConnectionTestPart2(reader.result);
+    }
+
+    reader.readAsText(fileInput.files[0]);	
+  }
+
+  function stripCert (cert) {
+    cert = cert.replace("-----BEGIN CERTIFICATE-----\n",'');
+    cert = cert.replace("\n-----END CERTIFICATE-----",'');
+    return cert;
+  }
+
+  function ldapConnectionTestPart2 (cert) {
+    var host     = $('#dex_connector_ldap_host').val()
+    var port     = $('#dex_connector_ldap_port').val()
+    var startTLS = $('#dex_connector_ldap_start_tls_true').parent().hasClass('active')
+    // Flip logic of #anonymous_bind_toggle since expanded div means anonymous binding is 'false'
+    var anonBind = ($('#anonymous_bind_toggle').attr('aria-expanded') === 'false')
+    var dn       = $('#dex_connector_ldap_bind_dn').val()
+    var pass     = $("#dex_connector_ldap_bind_pw[required='required']").val();
+    var baseDN   = $('#dex_connector_ldap_user_base_dn').val()
+    var filter   = $('#dex_connector_ldap_user_filter').val()
+    var paramArray = anonBind ? [host, port, cert, baseDN, filter] : [host, port, cert, dn, pass, baseDN, filter]
+
+    if (noneEmpty(paramArray)) { // Check if form inputs are filled in
+      $.post('/settings/ldap_test', {
+        host: host, port: port, start_tls: startTLS, cert: stripCert(cert),
+        anon_bind: anonBind, dn: dn, pass: pass, base_dn: baseDN, filter: filter
+      }).done(function (data) {
+        if (data.result.test_pass) renderValidity('Sucessfully validated connection to LDAP server', true)
```

Comment:
Could you add the curly braces back to this statement? This is a common sense style in the js community.

Reply:
Another improvement that would also increase the readability is to extract the callback to its own function. Like:

```js

// I'm bad at naming sometimes hehe
$.post('/settings/ldap_test', params)
.done(handleSuccessTest)
.fail(handleFailTest)

// ...

function handleSuccessTest(data) {
   // the if/else block here
}
```

That would simplify the method a lot.

Reply:
Vitor: Whether braces are used in this sense is, I believe, a personal preference thing. As the author of the PR I have chosen what I prefer and looks good to me.

If SUSE as a company, or CaaSP as a team has a defined code styleguide that I can see and is told to everyone, and everyone agrees to follow, then please point it out and I would be happy to follow it.

Pull requests are not the place to have discussions of personal code styling. You should be checking functionality to determine whether you approve, and possibly blocking if the appearance of the code is such that it is unclear or confusing.

It is not confusing with or without the braces. I will not add them simply because you like it better. That conversation is not one that belongs here.

Additionally, the code is perfectly readable with the 2 lines of "good/fail" within anonymous functions. Moving those out into separate functions does not, in my opinion, increase readability. I additionally will not move those out simply because you want it. There needs to a definitive value to doing so beyond your feelings about it. Present a logical argument for doing so and I will be happy to.

Convince the team as a whole to follow some definitive style and I will be happy to abide by it.